app/models/manageiq/providers/ansible_tower/automation_manager/credential/api_create.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::Credential
+  module ApiCreate
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      credential = manager.with_provider_connection do |connection|
+        connection.api.credentials.create!(params)
+      end
+
+      # Get the record in our database
+      # TODO: This needs to be targeted refresh so it doesn't take too long
+      EmsRefresh.queue_refresh(manager, nil, true) if !manager.missing_credentials? && manager.authentication_status_ok?
```

Comment:
If you have to return the id of the object that is created (and looks like you are) then the rhev refresh new vm is probably the right approach, just create the basic object kick off a refresh and return the new id.

Otherwise I'd love to put anything waiting for a refresh in a Task/Job state machine.